spec/lib/synapse/service_watcher_base_spec.rb

```diff
@@ -109,5 +109,26 @@ def remove_arg(name)
         expect(subject.backends).to eq(backends + default_servers)
       end
     end
+
+    context 'with label_filter set' do
+      labeled_backends = [
```

Comment:
this should be initialized in a `let`.

also, for more clarity, might be nice to do something like this:
```ruby
let(:matching_labled_backends) { [] }
let(:non_matching_labled_backends) { [] }
let(:non_labled_backends) { [] }
```

then you can do
```ruby
subject.send(:set_backends, matching_labled_backends + non_matching + non_labled)
expect(subject.backends).to contain_exactly(matching_labled_backends)
```